modules/nanointeractiveBidAdapter.js

```diff
@@ -0,0 +1,204 @@
+import * as utils from '../src/utils.js';
+import {config} from '../src/config.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import { getStorageManager } from '../src/storageManager.js';
+
+const BIDDER_CODE = 'nanointeractive';
+const END_POINT_URL = 'https://ad.audiencemanager.de';
+const SSP_PLACEMENT_ID = 'pid';
+const SSP_NETWORK_ID = 'nid';
+const NQ = 'nq';
+const NQ_NAME = 'name';
+const CATEGORY = 'category';
+const CATEGORY_NAME = 'categoryName';
+const SUB_ID = 'subId';
+const REF = 'ref'; const LOCATION = 'loc';
+
+const specFactory = () => {
+  let nanoNid = null;
+  let nanoPid = null;
+
+  const createSingleBidRequest = (bid, bidderRequest) => {
+    // Set these two variables that will be used later in the script
+    // There is probably a better way to do this
+    nanoNid = createSSPNetworkIdParam(bid);
+    nanoPid = createSSPPlacementId(bid);
+
+    const location = utils.deepAccess(bidderRequest, 'refererInfo.referer');
+    const origin = utils.getOrigin();
+
+    const data = {
+      [SSP_PLACEMENT_ID]: nanoPid,
+      [SSP_NETWORK_ID]: nanoNid,
+      [NQ]: [createNqParam(bid)],
+      [CATEGORY]: [createCategoryParam(bid)],
+      [SUB_ID]: createSubIdParam(bid),
+      [REF]: createRefParam(),
+      sizes: bid.sizes.map(value => value[0] + 'x' + value[1]),
+      bidId: bid.bidId,
+      cors: origin,
+      [LOCATION]: location,
+      lsUserId: getLsUserId()
+    };
+
+    if (bidderRequest && bidderRequest.gdprConsent) {
+      data.gdprConsent = bidderRequest.gdprConsent.consentString;
+      data.gdprApplies = (bidderRequest.gdprConsent.gdprApplies) ? '1' : '0';
+    }
+
+    return data;
+  }
+
+  const createSSPNetworkIdParam = bid => {
+    return bid.params[SSP_NETWORK_ID] ? bid.params[SSP_NETWORK_ID] : null;
+  }
+
+  const createSSPPlacementId = bid => {
+    return bid.params[SSP_PLACEMENT_ID] ? bid.params[SSP_PLACEMENT_ID] : null;
```

Comment:
This will apply to createSSPPlacementId and createSSPNetworkIdParam but you can simply use the following respectively:

`return bid.params[SSP_PLACEMENT_ID] || null;`

`return bid.params[SSP_NETWORK_ID] || null;`